app/javascript/controllers/turbo_frame_history_controller.js

```diff
@@ -22,9 +22,10 @@ export default class extends Controller {
     // This "fixes" history state for turbo frames, but it doesn't use the Turbo page cache.
     // This simply means that going back will cuase a new server request, but that's okay for our
     // purposes.
-    let fetchTargetingFrame = (event.detail.fetchOptions.headers['Turbo-Frame'] != undefined)
+    let getFetchTargetingFrame =  (event.detail.fetchOptions.headers['Turbo-Frame'] != undefined) &&
+                                  (event.detail.fetchOptions.method == "get")
```

Comment:
I figured out what the issue was. The POST to the form was getting added to the back state history. This extra check prevents that. I tested locally and it should be working now.